packages/toolpad-app/src/utils/collections.ts

```diff
@@ -31,20 +39,29 @@ export function mapProperties<U, V>(
   );
 }
 
+/**
```

Comment:
Doesn't hurt to have these descriptions, but i feel like a few (not all) of these utilities are almost self-explanatory and don't really need comments? So maybe we don't need to enforce that every utility method always has comments (it's fine to keep them now though), and it's ok to rely on self-explanatory method names?
Also have we decided definitively if we want to maintain our own utilities or rely in an external library? I'm fine with either (maybe slightly more in favor of using a library), but we should just find a consensus I guess.

Reply:
> Also have we decided definitively if we want to maintain our own utilities or rely on an external library (just for the typical stuff that for example `lodash` already does)? I'm fine with either (maybe slightly more in favor of using a library), but we should just find a consensus I guess.

We will just keep making informed decisions based on the use-case at hand, the available options, our experience, the maintenance cost required, etc... and not fall in the trap of taking on dogmas around this. Making the decision of using a library that does 85% of what you need may as well be detrimental to your overall codebase if the other 15% turn into a maintenance hell. Almost all of functions that are in the utils are there because of shortcomings in available options.